src/PdfSmith.BusinessLayer/Services/TimeZoneService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.AspNetCore.Http;
+using PdfSmith.BusinessLayer.Services.Interfaces;
+
+namespace PdfSmith.BusinessLayer.Services;
+
+public class TimeZoneService(IHttpContextAccessor httpContextAccessor) : ITimeZoneService
+{
+    public static readonly string X_TIME_ZONE = "x-time-zone";
+
+    public TimeZoneInfo? GetTimeZone()
+    {
+        var timeZoneId = GetTimeZoneHeaderValue();
+
+        if (timeZoneId is null)
+        {
+            return null;
+        }
+
+        return GetTimeZoneInfo(timeZoneId);
+    }
+
+    public string? GetTimeZoneHeaderValue()
+    {
+        if (httpContextAccessor.HttpContext?.Request?.Headers?.TryGetValue(X_TIME_ZONE, out var timeZone) ?? false)
+        {
+            return timeZone.ToString();
```

Comment:
What is the value of `timeZome `it you specify an header with an empty value?

`x-time-zone:  `

Reply:
it gives you an error because there isn't any time zone with id ""
I think the rule should be that if you specify the header, the value must be valid

Reply:
![image](https://github.com/user-attachments/assets/11637271-60c4-4d4c-add5-eceabdcebbba)
Just tested 😄

Reply:
you have passed the 'null' string as the value of the header. I mean do not specify a value at all:

`--header 'x-time-zone:  '`

Reply:
It works even with `--header 'x-time-zone:  '`